### DISCUSSION TITLE
Getting auth with HTTP GET not supported

### DISCUSSION THREAD

**Discussion by TheShadowGamer** [nextauthjs/next-auth#3107] [Help] [answered]
Whenever I try to access any `/api/auth` endpoint, I get the following error: `Action auth with HTTP GET is not supported by NextAuth.js`. The error seems fairly straight forward, but it occurs when I use the signIn() function from NextAuth.js so I'm not sure what to do. Any help would be appreciated!

**Comment by balazsorban44** [comment]
That endpoint does not exist. You can find the existing ones here: https://next-auth.js.org/getting-started/rest-api

A reproduction of your code would help resolve your problem.

**Comment by TheShadowGamer** [comment, marked as answer]
It turns out that I just had the [...nextauth].ts file in the root of the API folder and not in the api/auth folder, sorry 😅

**Reply by YoannBuzenet** [reply]
Same thing here haha

**Reply by ruohola** [reply]
And same here :D Thanks!

**Reply by Narottam04** [reply]
Same thing here lol

**Reply by assellalou** [reply]
seems like you are not the only one xD

**Reply by SteveSimms** [reply]
OMG_SAME!!! lol

**Reply by Kapaak** [reply]
Lolzi, same here :D

**Reply by JoakoV3** [reply]
OMG same 👍🏻

**Reply by johnrackles** [reply]
...same here

**Reply by johnstew** [reply]
Look at us 😅

**Reply by ChrisClaude** [reply]
same lol

**Reply by egkrly** [reply]
I was struggling for 2 days until saw your comment. I was about to give up on the library. :')

**Reply by Shahanees1998** [reply]
thanks dude

**Reply by divyanshutripathy** [reply]
I think we're a lot of people doing this

**Reply by deniivanov** [reply]
Yay, I'm not the only idiot :D spent an hour searching for the answer.

**Reply by AhmetEnesKCC** [reply]
> It turns out that I just had the [...nextauth].ts file in the root of the API folder and not in the api/auth folder, sorry 😅

thank you.

**Reply by shadeemerhi** [reply]
Same!!

**Reply by gvgray** [reply]
thanks

**Reply by GGAlanSmithee** [reply]
o dear ... thanks

**Reply by ghost** [reply]
Same...

**Reply by noam-elbaz** [reply]
Yup! Missed that too. Thanks.

**Reply by steniomoreira** [reply]
I'm part of this team!!

**Reply by simonsankar** [reply]
SAME THING

**Reply by ErlanBelekov** [reply]
so many of us 😅

**Reply by Akalanka-1996** [reply]
why why why😂🤦‍♂️

**Reply by 45w53525** [reply]
what to do now

**Reply by destinio** [reply]
wow Thank you. oh and I love you

**Reply by LucasSousa09** [reply]
same here, thank you very much

**Reply by jedusei** [reply]
Same here haha 😂😅

**Reply by ymaghzaz** [reply]
....oops , Thanks

**Reply by anhhtca** [reply]
Omg! Same problem 😄😄😄 I wasted a day trying to fix it until I read this issue.

**Reply by jamespearson** [reply]
Fixed it for me too, thanks for replying to your own issue when you solved it.

**Reply by engelmav** [reply]
Agreed. But something is ambiguous about the instructions that is leading us to this issue. Thanks to next-auth authors nonetheless for an amazing auth solution.

**Reply by wildskyf** [reply]
same here XD thanks for the answer!

**Reply by wysohn** [reply]
haha I knew I did something dumb. Thank you!

**Reply by sebastien-f** [reply]
I'll join the "didn't put it in the right spot" army :)

**Reply by ianhein** [reply]
Same here haha, thanks!

**Reply by JuanSebastianM** [reply]
joining the army too XD

**Reply by karimkawambwa** [reply]
🤦🏾

**Reply by drayilara** [reply]
Well, I am here to join the army. I am such a dummy lol

**Reply by AustinRyan** [reply]
I am also joining this army lol

**Reply by ccloveyy** [reply]
I am, too😂

**Reply by zaheeratwork** [reply]
Thought I was alone in this XD

**Reply by alexbatesdev** [reply]
The ShadowGamer stumbled and got back up so we could also stumble but then get back up as well, what a true hero

**Reply by RGSMTHLS** [reply]
This helped me alot

**Reply by DatNguyen2711** [reply]
oh my fucking god tks broooo :))))))))))))))

**Reply by AliceZ413** [reply]
haha, I struggled with it until I saw your answer, and then I realized I had done something dumb.

**Reply by rafaewaqarqazi** [reply]
Thank you 👍

**Reply by ezequielsalas** [reply]
Yeah, that was the cause

**Reply by Rishi2600** [reply]
still not able to figure it out using the latest approach, my file structure is `api\auth\[...nextauth]\route.ts`, can anyone please help me with it? I am able to use the `route.ts` normally but while authentication `Error: This action with HTTP GET is not supported by NextAuth.js` this is been showing still!

**Reply by teenu-s** [reply]
> still not able to figure it out using the latest approach, my file structure is `api\auth\[...nextauth]\route.ts`, can anyone please help me with it? I am able to use the `route.ts` normally but while authentication `Error: This action with HTTP GET is not supported by NextAuth.js` this is been showing still!

@Rishi2600 - did you get it resolved? I am also facing same issue

**Comment by KyleTryon** [comment]
For anyone else who found themselves here and _does_ have their `[...nextauth].ts` in the right place, this is what happened to me.

Make sure the URL you are hitting is correct. I was visiting `/api/auth/signIn` (which I think I saw documented somewhere) but the proper URL is shown here: https://next-auth.js.org/getting-started/rest-api, which is `/api/auth/signin`. Capitalization matters.

**Reply by FeMaffezzolli** [reply]
You may have seen here:
https://next-auth.js.org/getting-started/example#add-api-route

**Reply by ivangsp** [reply]
I had similar problem

**Reply by zaheeratwork** [reply]
@KyleTryon  - Thanks for this awesome. This is exactly the problem I had with document.

**Reply by AldinDrobic** [reply]
This solved it for me.

**Reply by neerajkashetty** [reply]
Thanks man!

**Reply by LBlend** [reply]
I love it when I can trust the documentation....

**Comment by B3none** [comment]
I've got the same issue as above but my `[...nextauth].ts` is in the correct place and it works correctly locally.

**Reply by GGAlanSmithee** [reply]
It's supposed to be `[...nextauth].ts`. not `[...nextauth.ts]`

**Reply by B3none** [reply]
Typo on my part, it’s in the correct place for me and is called the correct name - have corrected

**Comment by tibilodon** [comment]
Using Turbopack will result in the same error

**Reply by JustKira** [reply]
same when u went back to webpack it worked just fine i think its problem turbopack

**Reply by yanickrochon** [reply]
This error started to appear for no apparant reason. No error on the server, but the client fails `/api/auth/session` with that message. Everything is at the right place.

**Reply by jacobcoro** [reply]
> This error started to appear for no apparant reason. No error on the server, but the client fails `/api/auth/session` with that message. Everything is at the right place.

Please let me know if you resolve this as I'm having the same issue

**Reply by tyeongkim** [reply]
> > This error started to appear for no apparant reason. No error on the server, but the client fails `/api/auth/session` with that message. Everything is at the right place.
> 
> Please let me know if you resolve this as I'm having the same issue

same here

**Reply by jacobcoro** [reply]
Downgrading my versions of `next` and `next-auth` to `next@12.3.1` and `next-auth@4.14.0` solved this for me as per this suggestion: https://stackoverflow.com/a/74171248

**Comment by jacobcoro** [comment]
I've had this suddenly start appearing on my production site with no apparent change on my part. It works locally, and also works on my staging site which has the exact same code as production.

I tried the methods above to no avail. 

Please help!

**Reply by jacobcoro** [reply]
Downgrading my versions of `next` and `next-auth` to `next@12.3.1` and `next-auth@4.14.0` solved this for me as per this suggestion: https://stackoverflow.com/a/74171248

**Comment by nonfungible-dev** [comment]
I was running into this error in NextJS v13.0.6 until I downgraded next-auth to v4.16.0.

After downgrading, I could log in to all the services once again.

**Comment by goleary** [comment]
After upgrading to NextJS to v13.1.1 & NextAuth v 4.18.7 I am no longer having this error.

I was having problems while on on NextJS  v13.0.2 & NextAuth v4.18.4.

**Comment by allang** [comment]
I'm having this issue on Next 13.1.1 and NextAuth 4.21.1. It's a fresh Next app.

**Reply by tyeongkim** [reply]
I'm using Next 13.2.4 and Nextauth 4.20.1, then no issue.

**Comment by YuriyL-Git** [comment]
Could someone check this basic setup. Still getting this error on login button click. 
https://github.com/YuriyL-Git/test-auth-app

Setup is done according to official documentation.

**Comment by warrenday** [comment]
I had this issue when using GitHub for login. My callback URL was `https://www.mysite.com/api/auth/callback` however changing it to `https://mysite.com/api/auth/callback` fixed the issue. My env var for `NEXTAUTH_URL` was always `https://mysite.com`. So it was all due to a mismatch due to the extra www.

**Comment by pmioduszewski** [comment]
I'm using credentials provider (JWT + Django). My backend guy changed http -> https which I wasn't aware of. Adding "s" to the URL solved the issue.

**Comment by TimothyBrehm** [comment]
I had this issue while using App Router instead of Pages.  The issue turned out to be [...nextAuth] vs [...nextauth], which is boring, 
but the error message still references the pages architecture explicitly, which was a little confusing and obfuscated the issue.

**Comment by FtMr1** [comment]
By paying attention to the folder arrangement, I resolved the error in pages/api/auth/[...nextauth].ts

**Comment by espressom** [comment]
If you have correctly set up `app/api/auth/[...nextauth]/route.ts` in your app router and still encounter the above error, please make sure you have properly registered the endpoint as `/api/auth/signin`.
I am currently using the following versions:

next: `14.0.3`,
next-auth: `4.24.5`.

**Reply by vladimirhacker** [reply]
It didn't help still the same error:
"Error: This action with HTTP POST is not supported by NextAuth.js"

**Reply by marsidorowicz** [reply]
are you using middleware?

**Comment by marsidorowicz** [comment]
In next 13+ app router it can be having NEXTAUTH_URL while using middleware, delete key from .env should help u like in my case

**Reply by kachar** [reply]
Did it solve it for you? I'm using the beta channel and this got broken with the last update to `^5.0.0-beta.9`

The logs show:


```
web:dev: [auth][error] UnknownAction: Cannot parse action at /api/auth/signin .Read more at https://errors.authjs.dev#unknownaction
web:dev:     at parseActionAndProviderId (webpack-internal:///(rsc)/../../node_modules/@auth/core/lib/utils/web.js:90:49)
```

**Reply by marsidorowicz** [reply]
Yes, had same errors, also check
Middleware if you follow guidelines

**Comment by akash942** [comment]
for anyone who  is having the same error make sure that the folder is named [...nextauth] and not [nextauth] this will make sure that anything that comes after "auth" in 'yourUrl/api/auth/*' is routed to this folder and the route handler you defined runs. Spent like 4 hours on this :)

**Comment by eehhssaan** [comment]
Its just folder structure issue. I had pages/api/[nextauth].js (which was wrong). I resolved the issue by just changing the folder structure to pages/api/auth/[nextauth].js
